src/components/BlockActionsModal.tsx

```diff
@@ -0,0 +1,67 @@
+import { Text, TouchableOpacity } from 'react-native';
+import { View } from 'react-native';
+import Modal from 'react-native-modal';
+import React from 'react';
+import Ionicon from '@expo/vector-icons/Ionicons';
+
+interface BlockActionsModalProps {
+  deleteWorkout: () => void;
+  setModalVisible: (bool: boolean) => void;
+  renameWorkout: () => void;
+  copyWorkout: () => void;
+}
+const BlockActionsModal: React.FC<BlockActionsModalProps> = ({
+  deleteWorkout,
+  setModalVisible,
+  renameWorkout,
+  copyWorkout,
```

Comment:
I think we should keep this function name consistent with the name of the action. i.e., `duplicateWorkout` rather than copy, which could have a different meaning.